ivy/functional/ivy/data_type.py

```diff
@@ -2130,18 +2126,18 @@ def promote_types(
     query.sort(key=lambda x: str(x))
```

Comment:
I actually meant here that the way we're sorting the `query` isn't perfectly aligned with the order of [dtype pairs](https://github.com/unifyai/ivy/blob/47359a574231bd476f54fea63ee4fd8105d68991/ivy/__init__.py#L1291) in the promotion table, so if we fix the way we're sorting here or update the order in the promotion table if it's wrong anywhere, then we wouldn't need the try except at all, we can just do `return _promote(query)` from the function directly

Reply:
Done